### DISCUSSION TITLE
multiple format strings in clock.format

### DISCUSSION THREAD

**Discussion by genevieve-me** [Alexays/Waybar#2538] [Q&A] [answered]
I'd like to have two format strings for the clock module. However, when I try something like ``"clock": { "format": " {:%H:%M:%S} 󰸘 {:%F}",` I get `[error] clock: argument not found`, and the clock module is blank.

After a bit of trial and error, I found that `"clock": { "format": "{: %H:%M:%S 󰸘 %F}",` does work.
However, this leads waybar to constantly emit `[error] clock: invalid format string`. I was wondering what the correct way to configure this is and if this behavior is expected? Thanks!

**Comment by LukashonakV** [comment]
HI @genevieve-me ,
works well for me
```json
    "clock": {
"format": "{: %H:%M:%S 󰸘 %F}",
        "interval": 60,
        "format-alt": "{:%A, %B %d, %Y (%R) %Z}  ",
//        "tooltip-format": "<span size='12pt' font='Sarasa Mono SC'>{calendar}</span>",
        "tooltip-format": "<small><tt>{calendar}</tt></small>",
//        "tooltip-format": "{tz_list}<small><tt>{calendar}</tt></small>\n{:%Ec}",
        "timezones" : ["" , "Europe/Warsaw", "America/Chicago"],
        "calendar": {
                    "mode"          : "year",
                    "mode-mon-col"  : 3,
                    "weeks-pos"     : "right",
                    "on-scroll"     : 1,
                    "format": {
                              "months":     "<span color='#ffead3'><b>{}</b></span>",
                              "days":       "<span color='#ecc6d9'><b>{}</b></span>",
                              "weeks":      "<span color='#99ffdd'><b>W{:%U}</b></span>",
                              "weekdays":   "<span color='#ffcc66'><b>{}</b></span>",
                              "today":      "<span color='#ff6699'><b><u>{}</u></b></span>"
                              }
                    },
        "actions": {
                   "on-click-right": "mode",
                   "on-click-forward": "tz_up",
                   "on-click-backward": "tz_down",
                   "on-scroll-up": "shift_up",
                   "on-scroll-down": "shift_down"
                    },
    },

```

**Comment by LukashonakV** [comment]
Do you use standard clock or simple clock ? See [Wiki](https://github.com/Alexays/Waybar/wiki/Module:-Clock) about this

**Comment by mbme** [comment, marked as answer]
hey @genevieve-me and others who'll need it

you can use positional arguments to have multiple format strings:

`"clock": { "format": "{0:%H:%M:%S} {0:%F}" }`

Notice the `0` before the first semicolon in format string.

**Reply by ghost** [reply]
Why this doesn't work in temperature module?
Is it supposed to work?

**Reply by mbme** [reply]
the temperature module doesn't have any positional arguments, only [named arguments](https://github.com/Alexays/Waybar/wiki/Module%3A-Temperature#format-replacements) like `{temperatureC}` or `{icon}`
